File: backend/api/src/user/series.py
from database import db


class Series(db.Model):
    name = db.Column(db.String(64), unique=True, primary_key=True)
    user = db.Column(db.String(64))

    def __repr__(self):
        return "<Series {0} for User {1}>".format(self.name, self.user)

File: java-backend/src/main/java/com/anify/backend/model/Series.java
package com.anify.backend.model;

import jakarta.persistence.*;
import lombok.Data;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import java.util.List;

@Entity
@Data
public class Series {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    @OneToMany(mappedBy = "series", cascade = CascadeType.ALL)
    @JsonManagedReference
    private List<Song> songs;

    @OneToMany(mappedBy = "series", cascade = CascadeType.ALL)
    @JsonManagedReference
    private List<UserSeries> userSeries;
} 